src/core/input_descriptor.rs

```diff
@@ -0,0 +1,610 @@
+use ssi_dids::ssi_json_ld::syntax::from_value;
+
+/// A GroupId represents a unique identifier for a group of Input Descriptors.
+///
+/// This type is also used by the submission requirements to group input descriptors.
+pub type GroupId = String;
+
+/// A JSONPath is a string that represents a path to a specific value within a JSON object.
+///
+/// For syntax details, see [https://identity.foundation/presentation-exchange/spec/v2.0.0/#jsonpath-syntax-definition](https://identity.foundation/presentation-exchange/spec/v2.0.0/#jsonpath-syntax-definition)
+pub type JsonPath = String;
+
+/// The predicate Feature introduces properties enabling Verifier to request that Holder apply a predicate and return the result.
+///
+/// The predicate Feature extends the Input Descriptor Object `constraints.fields` object to add a predicate property.
+///
+/// The value of predicate **MUST** be one of the following strings: `required` or `preferred`.
+///
+/// If the predicate property is not present, a Conformant Consumer **MUST NOT** return derived predicate values.
+///
+/// See: [https://identity.foundation/presentation-exchange/#predicate-feature](https://identity.foundation/presentation-exchange/#predicate-feature)
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub enum Predicate {
+    /// required - This indicates that the returned value **MUST** be the boolean result of
+    /// applying the value of the filter property to the result of evaluating the path property.
+    #[serde(rename = "required")]
+    Required,
+    /// preferred - This indicates that the returned value **SHOULD** be the boolean result of
+    /// applying the value of the filter property to the result of evaluating the path property.
+    #[serde(rename = "preferred")]
+    Preferred,
+}
+
+/// Input Descriptors are objects used to describe the information a
+/// [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) requires of a
+/// [Holder](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:holder).
+///
+/// All Input Descriptors MUST be satisfied, unless otherwise specified by a
+/// [Feature](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:feature).
+///
+/// See: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-descriptor-object](https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-descriptor-object)
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq, Eq)]
+pub struct InputDescriptor {
+    id: String,
+    #[serde(default)]
+    constraints: Constraints,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    purpose: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    format: Option<ClaimFormatMap>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    group: Option<Vec<GroupId>>,
```

Comment:
Is it important to distinguish between `None` and an empty vec/map? If not:
```suggestion
    #[serde(default, skip_serializing_if = "ClaimFormatMap::is_empty")]
    format: ClaimFormatMap,
    #[serde(default, skip_serializing_if = "Vec::is_empty")]
    group: Vec<GroupId>,
```